File: backend/PersonalFinanceTracker.Api/Entities/Goal.cs
namespace PersonalFinanceTracker.Api.Entities;

public class Goal
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string UserId { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public decimal TargetAmount { get; set; }
    public decimal CurrentAmount { get; set; }
    public DateTime? TargetDate { get; set; }
    public string? CategoryId { get; set; }
    public string? LinkedAccountId { get; set; }
    public string? Icon { get; set; }
    public string? Color { get; set; }
    public string Status { get; set; } = "active";
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public AppUser? User { get; set; }
    public Category? Category { get; set; }
    public Account? LinkedAccount { get; set; }
    public List<TransactionRecord> Transactions { get; set; } = new();
}

File: frontend/src/types/goal.ts
export type Goal = {
  id: string;
  userId: string;
  name: string;
  targetAmount: number;
  currentAmount: number;
  remainingAmount: number;
  progressPercent: number;
  targetDate?: string | null;
  categoryId?: string | null;
  categoryName?: string | null;
  linkedAccountId?: string | null;
  linkedAccountName?: string | null;
  icon?: string | null;
  color?: string | null;
  status: "active" | "completed";
};

export type CreateGoalRequest = {
  name: string;
  targetAmount: number;
  targetDate?: string;
  categoryId?: string;
  linkedAccountId?: string;
  icon?: string;
  color?: string;
};

export type UpdateGoalRequest = CreateGoalRequest;

export type GoalActionRequest = {
  amount: number;
  accountId?: string;
  note?: string;
};
